### PR TITLE
[docs] Quantization

### PR DESCRIPTION
**Pull request by stevhliu**
A rough first draft of how quantization and PEFT can be used together. It mainly focuses on the QLoRA method and also references LoftQ.

### CODE REVIEW

File: docs/source/developer_guides/quantization.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+from peft import get_peft_model
+
+model = get_peft_model(model, config)
+```
+
+You're all set for training with whichever training method you prefer!
+
+### LoftQ initialization
+
+[LoftQ](https://hf.co/papers/2310.08659) initializes LoRA weights such that the quantization error is minimized. In contrast, PEFT initializes the weights from a Gaussian distribution by default. But LoftQ initialization can improve performance when training quantized models. To get started, create a [`LoftQConfig`] instead and set `loftq_bits=4` to for 4-bit quantization.
```
Comment (BenjaminBossan) [inline docs/source/developer_guides/quantization.md:97]: > In contrast, PEFT initializes the weights from a Gaussian distribution by default.

Not quite correct, default for LoRA is Kaiming uniform. But I think it doesn't really matter here, I'd probably remove the sentence completely (and adjust the start of the next one accordingly).

**Review by BenjaminBossan — APPROVED** [review]
Super good, thanks, only a small correction.

File: docs/source/developer_guides/quantization.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+rendered properly in your Markdown viewer.
+
+-->
+
+# Quantization
+
+Quantization represents data with fewer bits, making it a useful technique for reducing memory-usage and accelerating inference especially when it comes to large language models (LLMs). There are several ways to quantize a model including:
+
+* optimizing which model weights are quantized with the [AWQ](https://hf.co/papers/2306.00978) algorithm
+* independently quantizing each row of a weight matrix with the [GPTQ](https://github.com/PanQiWei/AutoGPTQ) algorithm
```
Comment (younesbelkada) [inline docs/source/developer_guides/quantization.md:22]:
```suggestion
* independently quantizing each row of a weight matrix with the [GPTQ](https://hf.co/papers/2210.17323) algorithm
```

**Review by younesbelkada — APPROVED** [review]
Very nice!

**Pull request merged**